### PR TITLE
Updated cc wrapper: switch from ld to vcheck if version is requested.

### PR DESCRIPTION
**Pull request by skosukhin**
A configure script might want to check the presence of a linker by requesting its version. The GNU linker ignores -rpath arguments if -v argument is given. This is not the case for Xcode (MacOS) linker, for example. The latter outputs its version and tries to process the rest of the arguments, which leads to a non-zero exit status because of the rpathes appended by the Spack's wrapper.

### CODE REVIEW

**Comment by citibeth** [comment]
So... this solves a problem on macOS?  In that case, I would want to see that it changes nothing when NOT running on macOS.

**Comment by skosukhin** [comment]
@citibeth, of course it's not a problem to add a check if we are on MacOS before switching to vcheck, but the question is: why should spack's wrapper act differently when it's requested for a compiler's version and for a linker's version? Currently when the wrapper acts as a compiler and gets the argument '-v', it calls the actual compiler and does not append additional arguments to the original set. It obviously does unnecessary (in terms of the compiler mode logic) work when it is called as a linker. Is it really how it should be?

My another question is: are the arguments '-v' and '-V' always about requesting a version when they are given to compilers (the other two from the same condition that activates the switch to the vcheck mode are self-explanatory)? If it is only an assumption then why can't we assume the same for linkers?

To sum up, I can make the fix that works only for macOS but I would like to understand why the logic should be different for compilers and linkers in this case (for example, it should be different for cpp because the argument '-v' is for verbose mode).

**Comment by citibeth** [comment]
I have not taken the time to understand the problem you are addressing in
detail.  However, I do know that seemingly innocuous changes can break
things.  The fewer such changes happen, the less likely things are to break.

Your logic may be sound and your conclusions correct.  But if you change
the way things work on Linux, your PR will have to be validated on Linux
before it can be merged.  There are many Linuxes and many compilers out
there, and testing on all of them could take a lot of effort.  Someone
would have to test for Cray too, which seems to do a lot of things a little
differently.

If you make your PR more narrowly focused to only change things on macOS,
then you will have an easier time getting merged.  You will just have to
attest that you have tested it on macOS, that it works for you, and that it
doesn't change behavior outside of macOS.  If someone else later runs into
the same problem on another OS, we can extend your solution to that OS as
well.

**Comment by tgamblin** [comment]
@citibeth:
> I have not taken the time to understand the problem you are addressing in
detail.

I don't want to detract from all the helpful reviews you're adding to Spack PRs, but I think it's important to consider PRs in detail and not scare off potential (or existing) contributors.  This PR addresses a technical issue with wrapping `ld`, and I suspect it is harmless on Linux. I also think if Linux `ld` ever got stricter with its args, like the Mac linker, this might actually be helpful on other platforms. 

I'm generally very careful with changes to `cc`, so if you don't have time to review this in detail you can be assured someone will before we merge it.  Also with the work @hegner has done recently with package testing will help in this regard -- we should have package build tests very soon.
